### PR TITLE
Theme Showcase: Use UpsellNudge component for Upwork banner

### LINKED ISSUE #54087: Theme Showcase: Update Experts banner use UpsellNudge component and styles
This banner appears in the current "advanced" themes section:

Screen Shot 2021-06-28 at 12 57 06 PM

Let's update the banner to use `UpsellNudge` rather than a custom component.

### PR DESCRIPTION
**Pull request by sixhours**
#### Changes proposed in this Pull Request

* Update current Upwork upsell to match existing UpsellNudges
* We may want to check in with marketing folks to make sure this change is OK from their end before we ship it.

**Before**

Screen Shot 2021-06-29 at 4 39 02 PM

**After**

Screen Shot 2021-06-29 at 4 48 21 PM

#### Testing instructions

* Switch to this PR, navigate to `/themes/[site]`
* Scroll to the bottom, click "Show all themes", you should see the new Upwork banner if you haven't dismissed it already
* Make sure the banner only appears to folks its intended to (need to double-check to see who that is)
* Make sure there are no visual bugs introduced
* Ensure notice can be dismissed
* Make sure all Tracks events (impression, click, dismiss) still work the same as before

Fixes #54087

### CODE REVIEW

File: client/blocks/upwork-banner/index.jsx [hunk 1]
```diff
@@ -79,33 +72,22 @@ class UpworkBanner extends PureComponent {
...
-					width={ 390 }
-					className="upwork-banner__image"
-					src={ builderIllustration }
+				<UpsellNudge
+					forceDisplay //Upwork banner has its own logic for showing/hiding
+					className="upwork-banner"
+					callToAction={ translate( 'Find your expert' ) }
+					onClick={ this.onStartNowClick }
+					href="https://wordpress.com/built-by-wordpress-com/"
+					title={ translate( 'Let our WordPress.com experts build your site!' ) }
```
Comment (a8ci18n) [inline client/blocks/upwork-banner/index.jsx:48]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Let Our WordPress.com Experts Build Your Site!' )` *ES Score: 11*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/upwork-nudge)

**Comment by sixhours** [comment]
Dismiss Tracks event is no longer working, I need to dig deeper into that tomorrow.

**Comment by kwight** [comment]
Looks great. I noticed there's a `calypso_upwork_banner_dismiss_icon_click` Tracks event when dismissing the original, but nothing for the new banner – can that be added into the upsell banner somehow (can be a followup)?

| Before | After |
| --- | --- |
| Screen Shot 2021-06-29 at 4 40 11 PM | Screen Shot 2021-06-29 at 4 40 39 PM |
| Screen Shot 2021-06-29 at 4 39 15 PM | Screen Shot 2021-06-29 at 4 38 04 PM |

**Review by kwight — APPROVED** [review]

**Comment by sixhours** [comment]
Refactored the Tracks events to fix the dismiss event not firing and consolidate everything into `UpsellNudge`. Tracks event names and props should be the same, although there may be extra props that should not affect tracking/stats. Look at all those red lines. 😍

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6105184).

**Comment by mreishus** [comment]
:+1: :+1: :+1: :+1: :+1: :+1: :+1: :+1: 
For some reason I keep accidentally clicking on the old one, the new one does not have that problem for me. :smile:

**Comment by sixhours** [comment]
Pinged Majorie in Slack to get a final go-ahead on this.

**Comment by sixhours** [comment]
Looks like I need to update a unit test, too.

**Comment by sixhours** [comment]
Got the go ahead, once the tests pass I will ship it :)

**Comment by kwight** [comment]
@sixhours I had to revert this in #54333 – @ianstewart noticed we've had a dropoff of clicks since deploying (p1625598429355600-slack-C0202PMUXLJ), and both he and I also failed to be forwarded to the URL when clicking the call-to-action. However, I'm _not_ able to replicate the failed redirect on prod with other upsell banners (like the sidebar domain upsell), nor on local dev with this original PR. So, just something to test and keep in mind when re-implementing.

**Pull request merged**